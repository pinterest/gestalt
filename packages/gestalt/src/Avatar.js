// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Box from './Box.js';
import Icon from './Icon.js';
import Image from './Image.js';
import Mask from './Mask.js';
import typography from './Typography.css';

const Square = (props: *) => (
  <Box {...props} position="relative">
    <Box
      dangerouslySetInlineStyle={{ __style: { paddingBottom: '100%' } }}
      position="relative"
    />
    <Box position="absolute" top left bottom right>
      {props.children}
    </Box>
  </Box>
);

const DefaultAvatar = ({ name }: { name: string }) => {
  const firstInitial = name ? [...name][0].toUpperCase() : '';
  return (
    <Square color="gray" shape="circle">
      {firstInitial && (
        <svg
          width="100%"
          viewBox="-50 -50 100 100"
          version="1.1"
          preserveAspectRatio="xMidYMid meet"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>{name}</title>
          <text
            fontSize="50px"
            fill="#fff"
            dy="0.35em"
            textAnchor="middle"
            className={[
              typography.antialiased,
              typography.sansSerif,
              typography.leadingSmall,
              typography.fontWeightBold,
            ].join(' ')}
          >
            {firstInitial}
          </text>
        </svg>
      )}
    </Square>
  );
};

type State = {| isImageLoaded: boolean |};

type AvatarProps = {|
  name: string,
  outline?: boolean,
  size?: 'sm' | 'md' | 'lg',
  src?: string,
  verified?: boolean,
  icon?: 'check-circle' | 'pinterest',
|};

const sizes = {
  sm: 24,
  md: 40,
  lg: 72,
};

export default class Avatar extends React.PureComponent<AvatarProps, State> {
  static propTypes = {
    name: PropTypes.string.isRequired,
    outline: PropTypes.bool,
    src: PropTypes.string,
    size: PropTypes.oneOf(['sm', 'md', 'lg']),
    verified: PropTypes.bool,
  };

  state = {
    isImageLoaded: true,
  };

  handleImageError = () => this.setState({ isImageLoaded: false });

  render() {
    const {
      name,
      outline,
      size,
      src,
      verified,
      icon = 'check-circle',
    } = this.props;
    const { isImageLoaded } = this.state;
    const width = size ? sizes[size] : '100%';
    const height = size ? sizes[size] : '';

    return (
      <Box
        color="white"
        {...(outline
          ? {
              dangerouslySetInlineStyle: {
                __style: {
                  boxShadow: '0 0 0 2px #fff',
                },
              },
            }
          : {})}
        width={width}
        height={height}
        position="relative"
        shape="circle"
      >
        {src && isImageLoaded ? (
          <Mask shape="circle" wash>
            <Image
              alt={name}
              color="#EFEFEF"
              naturalHeight={1}
              naturalWidth={1}
              src={src}
              onError={this.handleImageError}
            />
          </Mask>
        ) : (
          <DefaultAvatar name={name} />
        )}
        {verified && (
          <Box
            position="absolute"
            width="20%"
            height="20%"
            minWidth={8}
            minHeight={8}
            dangerouslySetInlineStyle={{
              __style: {
                bottom: '4%',
                right: '4%',
              },
            }}
          >
            <Box
              color="white"
              width="100%"
              height="100%"
              shape="circle"
              dangerouslySetInlineStyle={{
                __style: {
                  boxShadow: '0 0 0 2px #fff',
                },
              }}
            >
              <Icon color="red" icon={icon} accessibilityLabel="" size="100%" />
            </Box>
          </Box>
        )}
      </Box>
    );
  }
}
