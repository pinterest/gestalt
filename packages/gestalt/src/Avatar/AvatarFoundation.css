.container {
  border-radius: 50%;
  padding-bottom: 100%;
  position: relative;
}

.outline {
  border-radius: 50%;
  outline: 1px solid rgb(255 255 255);
}

.outlineVR {
  border-radius: var(--sema-rounding-circle);
  outline: 2px solid var(--sema-color-border-inverse);
}

.focused {
  border: 2px solid var(--sema-color-border-focus-inner-default);
  border-radius: 50%;
  outline: 2px solid var(--sema-color-border-focus-outer-default);
}

.imageHovered {
  background: var(--sema-color-hover-background-wash-image-tint);
}

.imagePressed {
  background: var(--sema-color-pressed-background-wash-image-tint);
}

.innerDiv {
  display: flex;
  inset: 0;
  justify-content: center;
  position: absolute;
}

.text {
  composes: antialiased sansSerif fontWeightSemiBold from "../Typography.css";
}

.vrText {
  composes: lg from "../TextUI.css";
}

.icon {
  stroke-width: 0;
  vertical-align: middle;
}

.iconFillDefault {
  fill: var(--color-icon-avatar-default);
}

.iconFillDarkMode {
  fill: var(--color-icon-light);
}
