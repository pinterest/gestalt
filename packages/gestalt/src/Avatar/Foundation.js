// @flow strict
import { type Node } from 'react';
import classnames from 'classnames';
import Box from '../Box.js';
import typography from '../Typography.css';
import { useColorScheme } from '../contexts/ColorSchemeProvider.js';
import icons from '../icons/index.js';
import styles from '../Icon.css';
import colors from '../Colors.css';
import avatarStyles from '../AvatarGroup.css';

const ICON_SIZE_RATIO = (20 / 48) * 100; // For pixel perfect icon button, we use the icon (20px) to parent container (48px) size ratio

type ResponsiveFitSizeBoxProps = {| children: Node, outline: boolean |};

function ResponsiveFitSizeBox({ children, outline }: ResponsiveFitSizeBoxProps): Node {
  const { colorGray0 } = useColorScheme();

  return (
    <Box
      color="secondary"
      dangerouslySetInlineStyle={{
        __style: {
          // When specifying a padding by percentage, it's always based on the width of the parent container so we get a property that's equal to the width.s
          paddingBottom: '100%',
          boxShadow: outline ? `0 0 0 1px ${colorGray0}` : undefined,
        },
      }}
      position="relative"
      rounding="circle"
    >
      <Box
        position="absolute"
        // top left bottom right constrains the circle to the exact dimensions of the responsive parent square
        top
        left
        bottom
        right
        display="flex"
        justifyContent="center"
      >
        {children}
      </Box>
    </Box>
  );
}

type Props = {|
  children?: string | number,
  fontSize?: string,
  outline?: boolean,
  textAnchor?: 'start' | 'middle' | 'end',
  title?: string,
  translate?: 'translateX10',
  content?: 'text' | 'icon',
|};

export default function AvatarFoundation({
  children,
  fontSize,
  outline = false,
  textAnchor = 'middle',
  title,
  translate,
  content = 'text',
}: Props): Node {
  const { colorGray300 } = useColorScheme();

  const cs = classnames(styles.icon, colors.darkGray);

  return (
    <ResponsiveFitSizeBox outline={outline}>
      {content === 'text' ? (
        <svg
          width="100%"
          viewBox="-50 -50 100 100"
          version="1.1"
          preserveAspectRatio="xMidYMid meet"
          xmlns="http://www.w3.org/2000/svg"
        >
          {title ? <title>{title}</title> : null}
          <text
            fontSize={fontSize}
            fill={colorGray300}
            dy="0.35em"
            textAnchor={textAnchor}
            className={[
              typography.antialiased,
              typography.sansSerif,
              typography.fontWeightSemiBold,
              translate && avatarStyles[translate], // if addCollaborator button is present, translateX moves numbers closer to the edge
            ].join(' ')}
          >
            {children}
          </text>
        </svg>
      ) : null}
      {content === 'icon' ? (
        <svg
          className={cs}
          width={`${ICON_SIZE_RATIO}%`} // percentual width to the parent container, reduces icon to 20px on a 48px parent container and keeps proportions upon resizing
          viewBox="0 0 24 24" // full icon size
          role="img"
          version="1.1"
          preserveAspectRatio="xMidYMid meet"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>Icon</title>
          <path d={icons.add} />
        </svg>
      ) : null}
    </ResponsiveFitSizeBox>
  );
}
