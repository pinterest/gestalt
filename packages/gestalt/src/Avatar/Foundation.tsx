import { ReactNode } from 'react';
import classnames from 'classnames';
import { TOKEN_COLOR_BORDER_AVATAR, TOKEN_COLOR_TEXT_DEFAULT } from 'gestalt-design-tokens';
import avatarStyles from '../AvatarGroup.css';
import Box from '../Box';
import styles from '../Icon.css';
import icons from '../icons/index';
import vrIcons from '../icons-vr-theme/index';
import typographyStyle from '../Typography.css';
import useInExperiment from '../useInExperiment';

const ICON_SIZE_RATIO = (20 / 48) * 100; // For pixel perfect icon button, we use the icon (20px) to parent container (48px) size ratio

type ResponsiveFitSizeBoxProps = {
  children: ReactNode;
  outline: boolean;
};

function ResponsiveFitSizeBox({ children, outline }: ResponsiveFitSizeBoxProps) {
  return (
    <Box
      color="secondary"
      dangerouslySetInlineStyle={{
        __style: {
          // When specifying a padding by percentage, it's always based on the width of the parent container so we get a property that's equal to the width.s
          paddingBottom: '100%',
          boxShadow: outline ? `0 0 0 1px ${TOKEN_COLOR_BORDER_AVATAR}` : undefined,
        },
      }}
      position="relative"
      rounding="circle"
    >
      <Box
        bottom
        // top left bottom right constrains the circle to the exact dimensions of the responsive parent square
        display="flex"
        justifyContent="center"
        left
        position="absolute"
        right
        top
      >
        {children}
      </Box>
    </Box>
  );
}

type Props = {
  children?: string | number;
  fontSize?: string;
  outline?: boolean;
  textAnchor?: 'start' | 'middle' | 'end';
  title?: string;
  translate?: 'translateX10';
  content?: 'text' | 'icon';
};

export default function AvatarFoundation({
  children,
  fontSize,
  outline = false,
  textAnchor = 'middle',
  title,
  translate,
  content = 'text',
}: Props) {
  const cs = classnames(styles.icon, avatarStyles.text);
  const isInExperiment = useInExperiment({
    webExperimentName: 'web_gestalt_visualRefresh',
    mwebExperimentName: 'web_gestalt_visualRefresh',
  });

  return (
    <ResponsiveFitSizeBox outline={outline}>
      {content === 'text' ? (
        <svg
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="-50 -50 100 100"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          {title ? <title>{title}</title> : null}
          <text
            className={[
              typographyStyle.antialiased,
              typographyStyle.sansSerif,
              typographyStyle.fontWeightSemiBold,
              translate && avatarStyles[translate], // if addCollaborator button is present, translateX moves numbers closer to the edge
            ].join(' ')}
            dy="0.35em"
            fill={TOKEN_COLOR_TEXT_DEFAULT}
            fontSize={fontSize}
            textAnchor={textAnchor}
          >
            {children}
          </text>
        </svg>
      ) : null}
      {content === 'icon' ? (
        <svg
          className={cs}
          preserveAspectRatio="xMidYMid meet" // percentual width to the parent container, reduces icon to 20px on a 48px parent container and keeps proportions upon resizing
          role="img" // full icon size
          version="1.1"
          viewBox="0 0 24 24"
          width={`${ICON_SIZE_RATIO}%`}
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>Icon</title>
          <path d={(isInExperiment ? vrIcons : icons)['person-add']} />
        </svg>
      ) : null}
    </ResponsiveFitSizeBox>
  );
}
