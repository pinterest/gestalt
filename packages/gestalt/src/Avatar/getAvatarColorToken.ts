interface ColorToken {
  base: string;
  hover: string;
  pressed: string;
}

interface ColorTokens {
  [key: string]: ColorToken;
}

// Need to replace color tokens
const colorTokensLight: ColorTokens = Object.freeze({
  'default': {
    base: '#E8E7E1',
    hover: '#D6D4CD',
    pressed: '#C2C1BC',
  },
  '01': {
    // redWeak
    base: '#FFD3D1',
    hover: '#FBBEBB',
    pressed: '#F4A8A4',
  },
  '02': {
    // redDefault
    base: '#FF9494',
    hover: '#F87777',
    pressed: '#ED5A5A',
  },
  '03': {
    // purpleWeak
    base: '#FCD8FC',
    hover: '#FAC7FA',
    pressed: '#F8B5F8',
  },
  '04': {
    // purpleDefault
    base: '#EFADEF',
    hover: '#E599E5',
    pressed: '#DA86DA',
  },
  '05': {
    // blueWeak
    base: '#C5EAF7',
    hover: '#A8DDF0',
    pressed: '#8ECFE6',
  },
  '06': {
    // blueDefault
    base: '#7CBEDE',
    hover: '#58A9D0',
    pressed: '#4096BF',
  },
  '07': {
    // greenWeak
    base: '#D0E2A8',
    hover: '#BCD090',
    pressed: '#A7B97E',
  },
  '08': {
    // greenDefault
    base: '#A8BB63',
    hover: '#93A550',
    pressed: '#81904C',
  },
  '09': {
    // orangeWeak
    base: '#FDE7C9',
    hover: '#F9D9AE',
    pressed: '#F4CC95',
  },
  '10': {
    // orangeDefault
    base: '#FBC55B',
    hover: '#F8B430',
    pressed: '#F1A613',
  },
});

const colorTokensDark: ColorTokens = Object.freeze({
  'default': {
    base: '#757570',
    hover: '#757570',
    pressed: '#757570',
  },
  '01': {
    // redWeak
    base: '#8A0F0F',
    hover: '#8A0F0F',
    pressed: '#8A0F0F',
  },
  '02': {
    // redDefault
    base: '#B2001A',
    hover: '#B2001A',
    pressed: '#B2001A',
  },
  '03': {
    // purpleWeak
    base: '#6D4270',
    hover: '#6D4270',
    pressed: '#6D4270',
  },
  '04': {
    // purpleDefault
    base: '#8F4696',
    hover: '#8F4696',
    pressed: '#8F4696',
  },
  '05': {
    // blueWeak
    base: '#215D82',
    hover: '#215D82',
    pressed: '#215D82',
  },
  '06': {
    // blueDefault
    base: '#007DB8',
    hover: '#007DB8',
    pressed: '#007DB8',
  },
  '07': {
    // greenWeak
    base: '#265926',
    hover: '#265926',
    pressed: '#265926',
  },
  '08': {
    // greenDefault
    base: '#517D3B',
    hover: '#517D3B',
    pressed: '#517D3B',
  },
  '09': {
    // orangeWeak
    base: '#9B4B1C',
    hover: '#9B4B1C',
    pressed: '#9B4B1C',
  },
  '10': {
    // orangeDefault
    base: '#C66F31',
    hover: '#C66F31',
    pressed: '#C66F31',
  },
});

type BackgroundColorLight = keyof typeof colorTokensLight;
type BackgroundColorDark = keyof typeof colorTokensDark;
type GetAvatarColorToken = (
  color: BackgroundColorLight | BackgroundColorDark,
  isHovered?: boolean,
  isPressed?: boolean,
  isDarkMode?: boolean,
) => string;

const getAvatarColorToken: GetAvatarColorToken = (color, isHovered, isPressed, isDarkMode) => {
  const colorToken = isDarkMode ? colorTokensDark[color] : colorTokensLight[color];

  if (!colorToken || !color) {
    throw new Error(`Invalid background color: ${color}`);
  }

  if (isPressed) {
    return colorToken.pressed;
  }

  if (isHovered) {
    return colorToken.hover;
  }

  return colorToken.base;
};

export default getAvatarColorToken;
