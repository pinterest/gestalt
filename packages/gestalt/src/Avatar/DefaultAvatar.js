// @flow strict
import { type Node } from 'react';
import AvatarFoundation from './Foundation.js';

type Props = {
  accessibilityLabel?: string,
  name: string,
};

export default function DefaultAvatar({ accessibilityLabel, name }: Props): Node {
  const firstInitial = name ? Array.from(name)[0].toUpperCase() : '';
  const title = accessibilityLabel ?? name;

  return (
    <AvatarFoundation fontSize="40px" textAnchor="middle" title={title}>
      {firstInitial}
    </AvatarFoundation>
  );
}
