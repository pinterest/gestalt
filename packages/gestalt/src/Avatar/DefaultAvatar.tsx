import AvatarFoundation from './Foundation';

type Props = {
  accessibilityLabel?: string;
  color?: '01' | '02' | '03' | '04' | '05' | '06' | '07' | '08' | '09' | '10';
  isHovered?: boolean;
  isPressed?: boolean;
  name: string;
};

export default function DefaultAvatar({
  accessibilityLabel,
  color,
  isHovered,
  isPressed,
  name,
}: Props) {
  const firstInitial = name ? Array.from(name)[0]?.toUpperCase() : '';
  const title = accessibilityLabel ?? name;

  return (
    <AvatarFoundation
      color={color}
      fontSize="40px"
      isHovered={isHovered}
      isPressed={isPressed}
      textAnchor="middle"
      title={title}
    >
      {firstInitial}
    </AvatarFoundation>
  );
}
