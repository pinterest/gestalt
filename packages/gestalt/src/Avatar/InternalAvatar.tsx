import { useState } from 'react';
import classnames from 'classnames';
import { TOKEN_COLOR_BACKGROUND_AVATAR_PLACEHOLDER } from 'gestalt-design-tokens';
import avatarStyles from './AvatarFoundation.css';
import DefaultAvatar from './DefaultAvatar';
import Box from '../Box';
import Icon from '../Icon';
import Image from '../Image';
import Mask from '../Mask';
import useInExperiment from '../useInExperiment';

const sizes = {
  xs: 24,
  sm: 32,
  md: 48,
  lg: 64,
  xl: 120,
} as const;

type Props = {
  accessibilityLabel?: string;
  color?: '01' | '02' | '03' | '04' | '05' | '06' | '07' | '08' | '09' | '10';
  isHovered?: boolean;
  isPressed?: boolean;
  name: string;
  outline?: boolean;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'fit';
  src?: string;
  verified?: boolean;
};

function InternalAvatar(props: Props) {
  const isInVRExperiment = useInExperiment({
    webExperimentName: 'web_gestalt_visualRefresh',
    mwebExperimentName: 'web_gestalt_visualRefresh',
  });
  const {
    accessibilityLabel,
    color,
    isHovered,
    isPressed,
    name,
    outline,
    size = isInVRExperiment ? 'md' : 'fit',
    src,
    verified,
  } = props;
  const [isImageLoaded, setIsImageLoaded] = useState(true);
  const handleImageError = () => setIsImageLoaded(false);
  const width = size === 'fit' ? '100%' : sizes[size];
  const height = size === 'fit' ? '' : sizes[size];

  return (
    <div
      className={classnames({
        [avatarStyles.outline]: !isInVRExperiment && outline,
        [avatarStyles.outlineVR]: isInVRExperiment && outline,
      })}
    >
      <Box
        data-test-id="gestalt-avatar-svg"
        height={height}
        position="relative"
        rounding="circle"
        width={width}
      >
        {src && isImageLoaded ? (
          <Mask rounding="circle" wash>
            <div
              className={classnames({
                [avatarStyles.imageHovered]: isInVRExperiment && isHovered,
                [avatarStyles.imagePressed]: isInVRExperiment && isPressed,
              })}
            >
              <Image
                alt={accessibilityLabel ?? name}
                color={TOKEN_COLOR_BACKGROUND_AVATAR_PLACEHOLDER}
                naturalHeight={1}
                naturalWidth={1}
                onError={handleImageError}
                src={src}
              />
            </div>
          </Mask>
        ) : (
          <DefaultAvatar
            accessibilityLabel={accessibilityLabel}
            color={color}
            isHovered={isHovered}
            isPressed={isPressed}
            name={name}
          />
        )}

        {verified && (
          <Box
            dangerouslySetInlineStyle={{
              __style: {
                bottom: '4%',
                right: '4%',
              },
            }}
            height="25%"
            minHeight={12}
            minWidth={12}
            position="absolute"
            width="25%"
          >
            <Box color="default" height="100%" rounding="circle" width="100%">
              <Icon accessibilityLabel="" color="brandPrimary" icon="check-circle" size="100%" />
            </Box>
          </Box>
        )}
      </Box>
    </div>
  );
}

export default InternalAvatar;
