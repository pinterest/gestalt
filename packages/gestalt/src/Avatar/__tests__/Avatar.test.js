// @flow
import React from 'react';
import { create } from 'react-test-renderer';
import Avatar from '../Avatar';

describe('Avatar', () => {
  it('renders multi-byte character initials', () => {
    const component = create(<Avatar name="ðŸ’© astral" />, {
      createNodeMock() {
        return { clientWidth: 100 };
      },
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with an empty name', () => {
    const tree = create(<Avatar name="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an outline', () => {
    const tree = create(<Avatar name="Jenny" outline />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the verified icon', () => {
    const tree = create(<Avatar name="Jammy" verified />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct src', () => {
    const tree = create(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct size - sm', () => {
    const tree = create(
      <Avatar
        name="Strava"
        src="http://pinterest.com/img/strave.png"
        size="sm"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct size - md', () => {
    const tree = create(
      <Avatar
        name="Strava"
        src="http://pinterest.com/img/strave.png"
        size="md"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct size - lg', () => {
    const tree = create(
      <Avatar
        name="Strava"
        src="http://pinterest.com/img/strave.png"
        size="lg"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
