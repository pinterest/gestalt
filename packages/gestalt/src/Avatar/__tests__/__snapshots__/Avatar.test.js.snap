// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders multibyte names 1`] = `
<div
  className="box circle relative whiteBg"
  style={
    Object {
      "boxShadow": "0 0 0 2px #fff",
      "height": "",
      "width": "100%",
    }
  }
>
  <div
    className="box circle grayBg relative"
  >
    <div
      className="box relative"
      style={
        Object {
          "paddingBottom": "100%",
        }
      }
    />
    <div
      className="absolute bottom0 box left0 right0 top0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        version="1.1"
        viewBox="-50 -50 100 100"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          ðŸ’© astral
        </title>
        <text
          className="antialiased sansSerif leadingSmall fontWeightBold"
          dominantBaseline="central"
          fill="#fff"
          fontSize="50px"
          textAnchor="middle"
        >
          ðŸ’©
        </text>
      </svg>
    </div>
  </div>
</div>
`;
